test/Bridges/Constraint/semi_to_binary.jl

```diff
@@ -0,0 +1,120 @@
+        (mock::MOIU.MockOptimizer) -> begin
+            MOI.set(mock, MOI.ObjectiveBound(), 0.0)
+            MOIU.mock_optimize!(mock, [0.0, 0.0, 0.0])
+        end,
+        (mock::MOIU.MockOptimizer) -> begin
+            MOI.set(mock, MOI.ObjectiveBound(), 2.0)
+            MOIU.mock_optimize!(mock, [2.0, 1.0, 1.0])
+        end,
+        (mock::MOIU.MockOptimizer) -> begin
+            MOI.set(mock, MOI.ObjectiveBound(), 2.0)
+            MOIU.mock_optimize!(mock, [2.0, 2.0, 1.0])
+        end,
+        (mock::MOIU.MockOptimizer) -> begin
+            MOI.set(mock, MOI.ObjectiveBound(), 2.5)
+            MOIU.mock_optimize!(mock, [2.5, 2.5, 1.0])
+        end,
+        (mock::MOIU.MockOptimizer) -> begin
+            MOI.set(mock, MOI.ObjectiveBound(), 3.0)
+            MOIU.mock_optimize!(mock, [3.0, 3.0, 1.0])
+        end,
+        (mock::MOIU.MockOptimizer) -> MOI.set(mock, MOI.TerminationStatus(), MOI.INFEASIBLE)
+    )
+    MOIT.semiconttest(bridged_mock,config)
+
+    ci = first(MOI.get(
+        bridged_mock,
+        MOI.ListOfConstraintIndices{MOI.SingleVariable,
+                                    MOI.Semicontinuous{Float64}}()))
+
+    test_delete_bridge(bridged_mock, ci, 2, (
+        (MOI.SingleVariable, MOI.EqualTo{Float64}, 1),
+        (MOI.SingleVariable, MOI.ZeroOne, 0),
+        (MOI.SingleVariable, MOI.Integer, 0),
+        (MOI.ScalarAffineFunction{Float64}, MOI.LessThan{Float64}, 0),
+        (MOI.ScalarAffineFunction{Float64}, MOI.GreaterThan{Float64}, 1),
+        ))
+
+    MOIU.set_mock_optimize!(mock,
+        (mock::MOIU.MockOptimizer) -> begin
+            MOI.set(mock, MOI.ObjectiveBound(), 0.0)
+            MOIU.mock_optimize!(mock, [0.0, 0.0, 0.0])
+        end,
+        (mock::MOIU.MockOptimizer) -> begin
+            MOI.set(mock, MOI.ObjectiveBound(), 2.0)
+            MOIU.mock_optimize!(mock, [2.0, 1.0, 1.0])
+        end,
+        (mock::MOIU.MockOptimizer) -> begin
+            MOI.set(mock, MOI.ObjectiveBound(), 2.0)
+            MOIU.mock_optimize!(mock, [2.0, 2.0, 1.0])
+        end,
+        (mock::MOIU.MockOptimizer) -> begin
+            MOI.set(mock, MOI.ObjectiveBound(), 3.0)
+            MOIU.mock_optimize!(mock, [3.0, 2.5, 1.0])
+        end,
+        (mock::MOIU.MockOptimizer) -> begin
+            MOI.set(mock, MOI.ObjectiveBound(), 3.0)
+            MOIU.mock_optimize!(mock, [3.0, 3.0, 1.0])
+        end,
+        (mock::MOIU.MockOptimizer) -> MOI.set(mock, MOI.TerminationStatus(), MOI.INFEASIBLE)
+    )
+    MOIT.semiinttest(bridged_mock,config)
+
+    ci = first(MOI.get(
+        bridged_mock,
+        MOI.ListOfConstraintIndices{MOI.SingleVariable,
+                                    MOI.Semiinteger{Float64}}()))
+
+    @test MOI.get(bridged_mock, MOI.ConstraintPrimal(), ci) == 3
+    new_set = MOI.Semiinteger{Float64}(19.0, 20.0)
+    MOI.set(bridged_mock, MOI.ConstraintSet(), ci, new_set)
+    @test MOI.get(bridged_mock, MOI.ConstraintSet(), ci) == new_set
+
```

Comment:
Add a test with `test_model_equals`, it will cover the missing getters for `ConstraintFunction` and `ConstraintSet`